test/e2e/tests/test_patchbaseline.py

```diff
@@ -0,0 +1,151 @@
+
+        update_data = {
+            "spec": {
+                "approvedPatches": ["KB5029928"]
+            }
+        }
+
+        k8s.patch_custom_resource(reference, update_data)
+        time.sleep(MODIFY_WAIT_AFTER_SECONDS)
+        assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "True", wait_periods=10)
+
+        updated_cr = k8s.get_resource(reference)
+        assert updated_cr["spec"]["approvedPatches"] == update_data["spec"]["approvedPatches"]
+        assert "approvalRules" in updated_cr["spec"]
+        assert "patchRules" in updated_cr["spec"]["approvalRules"]
+        assert len(updated_cr["spec"]["approvalRules"]["patchRules"]) == len(cr["spec"]["approvalRules"]["patchRules"])
+
+        baselineID = updated_cr["status"]["baselineID"]
+
+        # Check AWS API
+        patch_aws = ssm_client.get_patch_baseline(BaselineId=baselineID)
+
+        assert patch_aws is not None
+        assert patch_aws["BaselineId"] == baselineID
+        assert patch_aws["Name"] == updated_cr["spec"]["name"]
+        assert patch_aws["OperatingSystem"] == updated_cr["spec"]["operatingSystem"]
+        assert set(patch_aws["ApprovedPatches"]) == set(updated_cr["spec"]["approvedPatches"])
+
+        # Check approval rules
+        assert "ApprovalRules" in patch_aws
+        assert "PatchRules" in patch_aws["ApprovalRules"]
+        for aws_rule, cr_rule in zip(patch_aws["ApprovalRules"]["PatchRules"], updated_cr["spec"]["approvalRules"]["patchRules"]):
+            assert aws_rule["ApproveAfterDays"] == cr_rule["approveAfterDays"]
+            assert aws_rule["ComplianceLevel"] == cr_rule["complianceLevel"]
+            aws_filters = aws_rule["PatchFilterGroup"]["PatchFilters"]
+            cr_filters = cr_rule["patchFilterGroup"]["patchFilters"]
+            for aws_filter, cr_filter in zip(aws_filters, cr_filters):
+                assert aws_filter["Key"] == cr_filter["key"]
+                assert set(aws_filter["Values"]) == set(cr_filter["values"])
+
+        # Check Tags
+        tag_data = {
+            "spec": {
+                "tags": [
+                    {"key": "Environment", "value": "Production"},
+                    {"key": "Team", "value": "DevOps"}
+                ]
+            }
+        }
+
+        k8s.patch_custom_resource(reference, tag_data)
+        time.sleep(MODIFY_WAIT_AFTER_SECONDS)
+        assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "True", wait_periods=10)
+        
+        updated_cr = k8s.get_resource(reference)
+        assert "tags" in updated_cr["spec"]
+        assert updated_cr["spec"]["tags"] == tag_data["spec"]["tags"]
+
+        tags_aws = ssm_client.list_tags_for_resource(
+            ResourceType="PatchBaseline",
+            ResourceId=baselineID
+        )["TagList"]
+
+
+        # default tags: services.k8s.aws/controller-version, services.k8s.aws/namespace
+        assert len(tags_aws) - 2 == len(tag_data["spec"]["tags"])
```

Comment:
nit: we have a library to deal with ACK default labels https://github.com/aws-controllers-k8s/test-infra/blob/main/src/acktest/tags.py

Reply:
Done!